cdap-app-fabric/src/main/java/co/cask/cdap/config/ConfigStore.java

```diff
@@ -33,6 +33,14 @@
   void create(String namespace, String type, Config config) throws ConfigExistsException;
 
   /**
+   * Create or update a Configuration.
+   * @param namespace namespace
+   * @param type configuration type
+   * @param config configuration object
+   */
+  void createOrUpdate(String namespace, String type, Config config);
```

Comment:
yes it is needed. for example, for dashboards we want to throw an exception when we try to update a dashboard that doesn't exist. but in future when we have preferences stored at namespace, instance level etc, we want to be able to create or update preferences (and not throw an exception).